compiler/noirc_evaluator/src/acir/acir_context/mod.rs

```diff
@@ -840,6 +840,13 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
             } else {
                 bit_size - max_rhs_bits + 1
             };
+
+            // If the RHS is a result of an underflow, it could be a negative number which is represented by
+            // a very large positive Field, which could fail to compile to ACIR in `range_constrain_var` below,
+            // because it can use all 254 bits. To avoid that, ignore even what we know about the RHS and trust
+            // that a previous range constraint will fail at runtime before it comes to executing this instruction.
+            let max_rhs_bits = max_rhs_bits.min(bit_size);
```

Comment:
I think I understand the issue with underflowing, but I would expect a fix to be elsewhere and I would expect to bubble up this "will always fail at runtime" result as a warning (like we do with `assert(false)`.

How did this cause "flakiness in aztec-packages in http://ci.aztec-labs.com/list/ee2aaf4f842c6c24"? If this was e.g. panicking nondeterministically, I would expect it to be due to either 1) the `aztec-packages` test setup generating invalid inputs, or 2) non-determinism in Noir itself. I'm concerned that such potential bugs could be covered up by skipping this check.

Also, how can we tell at this point whether it's due to such an underflow (which we expect to check at runtime) or due to a different issue?

Reply:
But, using bit_size is not the end of the world, as that will happen for any non-constant value anyway, which can also over/underflow at runtime

Reply:
@michaeljklein ti be clear Aztec-packages runs all Noir tests, including the AST fuzzer smoke tests. It just happens to have failed for them, not us.

regarding a fix elsewhere: I looked at it last time; where the binary instruction is handled it just does it on Fields, and the adds an overflow check. It does not care if it’s constant or not. Only here do we look at it. I didn’t want to rewrite how that works, as it seems to be fine for Brillig, only ACIR gen assumes stuff about constants.

Reply:
Changed it to include an assertion as @guipublic suggested.